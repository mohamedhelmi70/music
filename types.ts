export type AppParamList = {
    Tab: undefined;
    Home: undefined;
    Track: {
        trackId: number;
    }
    Artists: undefined;
    Artist: {
        artistId: number;
    }
    Search: undefined;
    Albums: undefined;
    Album:  {
        albumId: number;
    }
};

export interface Album {
    album: string,
    id_album: number,
    cover: string,
    api_album: string,
    api_tracks: string
}

export interface Track {
    id_track: number;
    track: string;
    bpm: number;
    api_track: string;
    api_lyrics: string;
    artist: string;
}

export interface Artist {
    id_artist: number,
    artist: string,
    cover: string,
}