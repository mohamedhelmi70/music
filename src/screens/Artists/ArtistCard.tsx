import React from 'react';
import {StyleSheet, Pressable} from 'react-native';
import {Artist, AppParamList} from '../../../types';
import {StackNavigationProp} from '@react-navigation/stack';
import {wp} from '../../constants';
import FastImage from '../../components/FastImage';
import {Text, View} from '../../components/Themed';

interface ArtistCardProps {
    artist: Artist;
    navigation: StackNavigationProp<AppParamList, 'Artists'>;
}

function AlbumCard({artist, navigation}: ArtistCardProps) {
    return (
        <Pressable onPress={() => navigation.navigate('Artist', {artistId: artist.id_artist})} style={styles.container}>
            <View>{Boolean(artist.cover) && <FastImage url={artist.cover} style={styles.image} />}</View>
            <View style={styles.cardRight}>
                <Text style={styles.cardTitle} weight="700" numberOfLines={1} lightColor="#334443">
                    {artist.artist}
                </Text>
            </View>
        </Pressable>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: wp('45%'),
        borderRadius: wp('2%'),
        alignItems: 'center',
        marginBottom: wp('5%'),
    },
    image: {
        width: wp('40%'),
        height: wp('40%'),
        borderRadius: wp('20%'),
        marginBottom: wp('2%'),
    },
    cardRight: {
        marginLeft: wp('4%'),
    },
    cardTitle: {
        fontSize: wp('3%'),
        maxWidth: wp('40%'),
    },
});

export default AlbumCard;
