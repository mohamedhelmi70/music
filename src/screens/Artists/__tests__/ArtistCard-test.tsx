import React from 'react';
import renderer from 'react-test-renderer';
import {Artist} from '../../../../types';
import ArtistCard from '../ArtistCard';

// Describing a test suite
describe('<ArtistCard />', () => {
    const artist: Artist = {
        id_artist: 11,
        artist: 'artist name',
        cover: 'https://artist.png',
    };
    test('Renders ArtistCard correctly', () => {
        const tree = renderer.create(<ArtistCard artist={artist} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
