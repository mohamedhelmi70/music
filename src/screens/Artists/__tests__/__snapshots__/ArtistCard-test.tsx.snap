// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistCard /> Renders ArtistCard correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 15,
      "flex": 1,
      "marginBottom": 37.5,
      "width": 337.5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F7F7F9",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 150,
            "height": 300,
            "marginBottom": 15,
            "width": 300,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "cache": "immutable",
            "headers": Object {
              "Authorization": "4f298a53e552283bee957836a529baec",
            },
            "priority": "normal",
            "uri": "https://artist.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F7F7F9",
        },
        Object {
          "marginLeft": 30,
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#334443",
            "fontFamily": "Roboto",
            "fontWeight": "700",
            "textAlign": "left",
          },
          Object {
            "fontSize": 22.5,
            "maxWidth": 300,
          },
        ]
      }
    >
      artist name
    </Text>
  </View>
</View>
`;
